packages/babel-parser/src/plugins/typescript.js

```diff
@@ -219,7 +219,25 @@ export default (superClass: Class<Parser>): Class<Parser> =>
     }
 
     tsParseEntityName(allowReservedWords: boolean): N.TsEntityName {
-      let entity: N.TsEntityName = this.parseIdentifier();
+      let entity: N.TsEntityName;
+      // TODO: package this into its own tsParse function
+      if (this.state.type === tt._import) {
+        entity = this.startNode();
+        this.next();
+        if (!this.match(tt.parenL)) {
+          this.unexpected(null, tt.parenL);
+        }
+        this.eat(tt.parenL);
+        const args = this.parseCallExpressionArguments(tt.parenR, false, {
```

Comment:
I think `StringLiteral` is only allowed, because `import()` in type annotation is not a real function.